### ISSUE TITLE
abbreviation filter doesn't work without spaces

### ISSUE THREAD

**Issue by wenlzhang** [retorquere/zotero-better-bibtex#1216]
Hei,

When trying to generating the Citekey with the Citation key format `[auth][authForeIni][year][shorttitle3_3:abbr]`, it did not give the expected result.

From my understanding, the expected Citekey should consist of the last name of the first author, the initial of the forename of the first author, year, the initial of the first three letters of the title. For example, the key should be TchamnaR2013YRS. Instead, I got TchamnaR2013Y, which is a little strange.

Thanks in advance.

**Comment by wenlzhang** [comment]
Hi, the report ID is W576SACF-euc.

**Comment by retorquere** [comment]
I'll need you to right-click the reference that has `TchamnaR2013YRS` and generate a debug log from that; for instructions, see @label-gun's comment above.

**Comment by wenlzhang** [comment]
Hi,

The updated report ID is USRPVWPA-euc.

Thanks.

**Comment by retorquere** [comment]
`shorttitle3_3` yields `YawRateSideslip` for that reference, which has no spaces, so `abbr` thinks it's a single word, so it takes only the first letter of that. The default functions aim to simplify a lot so they're not very flexible; you may want something along the lines of

```
[auth][authForeIni][year][Title:skipwords:select=1,3:capitalize:abbr]
```

note the capital `Title`, that's the key; see also https://retorque.re/zotero-better-bibtex/citing/#filters and https://retorque.re/zotero-better-bibtex/citing/#direct-access-to-unprocessed-fields

**Comment by retorquere** [comment]
I'll see if I can change `shorttitle` to work with `abbr` without doing damage to behavior people have come to rely on.

**Comment by blip-bloop** [comment]
:robot: this is your friendly neighborhood build bot announcing [test build 5.1.98.4267](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.1.98.4267.xpi) ("util/site-update")

Install in Zotero by downloading [test build 5.1.98.4267](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.1.98.4267.xpi), opening the Zotero "Tools" menu, selecting "Add-ons", open the gear menu in the top right, and select "Install Add-on From File...".

**Comment by retorquere** [comment]
Can you try 4267 with `auth][authForeIni][year][shorttitle3_3:abbr]`?

**Comment by wenlzhang** [comment]
Hi,

Thanks for the quick response.

I tested with a few items, and both this new fix and your previous solution regarding "Title" worked for me.

**Comment by retorquere** [comment]
Cool, thanks for the confirmation -- I've released 5.1.99, but it's really just a rebadged 4267. You can stay on that version and you'll be updated automatically by Zotero in the next few days.
